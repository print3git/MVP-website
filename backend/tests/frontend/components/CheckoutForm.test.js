/** @jest-environment jsdom */
const React = require("react");
const { render } = require("@testing-library/react");
const fs = require("fs");
const path = require("path");
const babel = require("@babel/core");
jest.mock("react-places-autocomplete", () => {
  const React = require("react");
  return {
    __esModule: true,
    default: () => React.createElement("div"),
    geocodeByAddress: jest.fn(),
  };
});

const src = fs.readFileSync(
  path.join(__dirname, "../../../../src/components/CheckoutForm.js"),
  "utf8",
);
const { code } = babel.transformSync(src, {
  presets: [["@babel/preset-react", { runtime: "automatic" }]],
  plugins: [
    ["@babel/plugin-syntax-typescript", { isTSX: true }],
    "@babel/plugin-transform-modules-commonjs",
  ],
  filename: "CheckoutForm.js",
});
const Module = require("module");
const m = new Module("CheckoutForm.js");
m.paths = Module._nodeModulePaths(__dirname);
m._compile(code, "CheckoutForm.js");
const CheckoutForm = m.exports.default;

describe("CheckoutForm", () => {
  test.skip("renders consistently", () => {
    const { container } = render(React.createElement(CheckoutForm));
    expect(container).toMatchSnapshot();
  });
});
