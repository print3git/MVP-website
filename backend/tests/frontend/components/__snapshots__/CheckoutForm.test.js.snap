// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CheckoutForm renders consistently 1`] = `
<div>
  <form>
    <div>
      <div />
      <a
        href="#"
      >
        Can't find your address? Enter it manually
      </a>
    </div>
    <input
      name="street"
      type="hidden"
      value=""
    />
    <input
      name="city"
      type="hidden"
      value=""
    />
    <input
      name="zip"
      type="hidden"
      value=""
    />
    <input
      name="country"
      type="hidden"
      value=""
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
